ckanext/datarequests/actions.py

```diff
@@ -562,8 +573,9 @@ def close_datarequest(context, data_dict):
         raise tk.ValidationError([tk._('This Data Request is already closed')])
 
     data_req.closed = True
-    data_req.accepted_dataset_id = data_dict.get('accepted_dataset_id', None)
+    data_req.accepted_dataset_id = data_dict.get('accepted_dataset_id') or None
```

Comment:
Won't this throw an exception if the key is not present? The previous syntax seems more robust.

Reply:
since we want None anyway, this code just makes it more obvious, in the spec https://docs.python.org/2/library/stdtypes.html#dict.get if the second variable is not set, then None is the default so doing the or check would just set the same variable (from what the docs state) 
of course we could remove the ```or None``` and create a unit test to verify that it will works as expected  (or just rerun the current set)

Reply:
The `or` command uses boolean logic, so if `data_dict.get('accepted_dataset_id')` has no key present or the value is '' or None, the boolean value will be false and the value to the right of the 'or' command will be used. https://realpython.com/python-or-operator/
I had to use this because `data_dict.get('accepted_dataset_id', None)` would return an empty string if the key was present with no value. Where the correct value should be None.
It is pretty much a shorthand equivalent of `data_req.accepted_dataset_id = data_dict.get('accepted_dataset_id') if data_dict.get('accepted_dataset_id') and data_dict.get('accepted_dataset_id') != '' else None`

Reply:
cool, handing empty string.